libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -97,309 +97,416 @@ const AP_Param::GroupInfo AC_AttitudeControl::var_info[] = {
     AP_GROUPEND
 };
 
-void AC_AttitudeControl::relax_bf_rate_controller()
+// Set output throttle and disable stabilization
+void AC_AttitudeControl::set_throttle_out_unstabilized(float throttle_in, bool reset_attitude_control, float filter_cutoff)
+{
+    _throttle_in = throttle_in;
+    _motors.set_throttle_filter_cutoff(filter_cutoff);
+    if (reset_attitude_control) {
+        relax_attitude_controllers();
+    }
+    _motors.set_throttle(throttle_in);
+    _angle_boost = 0.0f;
+}
+
+// Ensure attitude controller have zero errors to relax rate controller output
+void AC_AttitudeControl::relax_attitude_controllers()
 {
+    // TODO add _ahrs.get_quaternion()
+    _attitude_target_quat.from_rotation_matrix(_ahrs.get_rotation_body_to_ned());
+    _attitude_target_ang_vel = _ahrs.get_gyro();
+    _attitude_target_euler_angle = Vector3f(_ahrs.roll, _ahrs.pitch, _ahrs.yaw);
+
     // Set reference angular velocity used in angular velocity controller equal
     // to the input angular velocity and reset the angular velocity integrators.
     // This zeros the output of the angular velocity controller.
-    _ang_vel_target_rads = _ahrs.get_gyro();
+    _rate_target_ang_vel = _ahrs.get_gyro();
     get_rate_roll_pid().reset_I();
     get_rate_pitch_pid().reset_I();
     get_rate_yaw_pid().reset_I();
-
-    // Write euler derivatives derived from vehicle angular velocity to
-    // _att_target_euler_rate_rads. This resets the state of the input shapers.
-    ang_vel_to_euler_rate(Vector3f(_ahrs.roll,_ahrs.pitch,_ahrs.yaw), _ang_vel_target_rads, _att_target_euler_rate_rads);
 }
 
-void AC_AttitudeControl::shift_ef_yaw_target(float yaw_shift_cd)
-{
-    _att_target_euler_rad.z = wrap_2PI(_att_target_euler_rad.z + radians(yaw_shift_cd*0.01f));
+// The attitude controller works around the concept of the desired attitude, target attitude
+// and measured attitude. The desired attitude is the attitude input into the attitude controller
+// that expresses where the higher level code would like the aircraft to move to. The target attitude is moved
+// to the desired attitude with jerk, acceleration, and velocity limits. The target angular velocities are fed
+// directly into the rate controllers. The angular error between the measured attitude and the target attitude is
+// fed into the angle controller and the output of the angle controller summed at the input of the rate controllers.
+// By feeding the target angular velocity directly into the rate controllers the measured and target attitudes
+// remain very close together.
+//
+// All input functions below follow the same procedure
+// 1. define the desired attitude the aircraft should attempt to achieve using the input variables
+// 2. using the desired attitude and input variables define the target angular velocity to that should
```

Comment:
Maybe it's me since English isn't my native language, but the sentence in number 2 doesn't sense to me. My previous comment was a suggestion to change _**to that**_ to _**so that it**_